### PR TITLE
Add `config` package and implement `ackdev edit/get config` sub commands

### PR DESCRIPTION
**Pull request by a-hilaly**
Issue: N/A

Description:
- Implement `ackdev get config` command
- Implement `ackdev edit config` command
- Add a `config` package that contains methods to load and save ackdev configuration

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by a-hilaly** [comment]
Need rebase after #3 merge

**Comment by a-hilaly** [comment]
@jaypipes To implement `ackdev ensure deps` we will need:
- Access to `ackdev` configuration file (implemented in this PR)
- A `repository` package exposing functionalities to manage local/github repositories (forks, local repo paths etc...)

Next PR i'll add the `repository` package and implement `ackdev ensure deps` after that

File: cmd/ackdev/cmd/config.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+func init() {
+	configCmd.AddCommand(editConfigCmd)
+	configCmd.AddCommand(viewConfigCmd)
+}
+
+var configCmd = &cobra.Command{
+	Use:   "config",
+	Args:  cobra.NoArgs,
+	Short: "View or edit ackdev configuration file",
+}
```
Comment (jaypipes) [inline cmd/ackdev/cmd/config.go]: I think we were going to go with `ackdev <verb> <noun>`, right? So, this should be `ackdev show config` (or the existing `ackdev list config`?) and then `ackdev edit config`.

File: cmd/ackdev/cmd/config_edit.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+import (
+	"os"
+	"os/exec"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultEditor = "vim"
```
Comment (jaypipes) [inline cmd/ackdev/cmd/config_edit.go]: use `vi`, not `vim` for the default... :)

File: cmd/ackdev/cmd/config_view.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+	"github.com/aws-controllers-k8s/dev-tools/pkg/config"
+)
+
+var (
+	optViewConfigOutput string
+)
+
+func init() {
+	viewConfigCmd.PersistentFlags().StringVarP(&optViewConfigOutput, "output", "o", "yaml", "output format (json|yaml)")
```
Comment (jaypipes) [inline cmd/ackdev/cmd/config_view.go]: And I would add this arg to https://github.com/aws-controllers-k8s/dev-tools/blob/3187ceb3b3d6551405fa33d57a75b86ab11b71e5/cmd/ackdev/cmd/list.go#L22 instead.

**Review by jaypipes — CHANGES_REQUESTED** [review]

File: cmd/ackdev/cmd/config.go (hunk 1 above)
Comment (a-hilaly) [reply cmd/ackdev/cmd/config.go]: Good point! changed to `ackdev list config`. The `list` sub command have also `get` as alias

File: cmd/ackdev/cmd/config_edit.go (hunk 2 above)
Comment (a-hilaly) [reply cmd/ackdev/cmd/config_edit.go]: Yikes! fixed :)

File: cmd/ackdev/cmd/config_view.go (hunk 3 above)
Comment (a-hilaly) [reply cmd/ackdev/cmd/config_view.go]: Very good point! Moved `--ouput` flag to the list command.

**Review by jaypipes — APPROVED** [review]
:+1:

**Pull request merged**